### DISCUSSION TITLE
Creating weather effects with a separate light and shadow group

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#204] [Q&A] [answered]
Something that I previously tried to do with Eevee and a janky refraction plane setup is now properly possible with Malt: post-processing weather effects!
![image](https://user-images.githubusercontent.com/88381426/146091998-578d4557-f9a1-49ff-88d0-7275583bf00f.png)
This is how far I've come already. This works great but ive not quite got the mask for where the 'rain haze' should be down. What I wanted to do to help with this is using the shading information of a sun light pointing straight down to mask the areas where rain hits the objects (where the benifit would be the light occlusion equalling the 'downfall occlusion'). I tried the 'scene_diffuse_gradient' node which works well but for some reason I only get cast shadows for the light group 1. Is there a way to also get proper cast shadows for any other light group? Below you can see that there are no cast shadows when using the light group 2:
![image](https://user-images.githubusercontent.com/88381426/146092505-f7476f1c-9b1b-4e2a-ab06-6980eee054d4.png)

**Comment by pragma37** [comment, marked as answer]
For materials to cast shadows in a Light Group you need to add the group into Light Groups > Shadow.

So, in other words, your current Shadow Groups is set to (1,0,0,0) and it should be (1,2,0,0).
